internal/artifacts/module_arch.go

```diff
@@ -129,6 +129,10 @@ func generateMtad(mtaObj *mta.MTA, loc dir.ITargetPath, target string,
 	}
 	mtadLocation := mtadLoc{path: mtadTargetPath}
 
+	removeUndeployedModules(mtaObj, platform)
+
+	setPlatformSpecificParameters(mtaObj, platform)
+
```

Comment:
These functions are already called from `generateMeta`. Can we maybe change the order a bit so they will be inside `genMtad` (and won't be called twice in the `generateMeta` flow)? Maybe generate the mtad.yaml before the manifest.mf?